File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/C/s041542819_commented.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_N 100005
#define INF 10000000000LL

typedef struct {
    FILE *in;
} LightScanner;

typedef struct {
    FILE *out;
} LightWriter;

LightScanner* LightScanner_new(FILE *in) {
    LightScanner *scanner = (LightScanner*)malloc(sizeof(LightScanner));
    scanner->in = in;
    return scanner;
}

char* LightScanner_string(LightScanner *scanner) {
    static char buffer[1024];
    fscanf(scanner->in, "%s", buffer);
    return buffer;
}

int LightScanner_ints(LightScanner *scanner) {
    return atoi(LightScanner_string(scanner));
}

long long LightScanner_longs(LightScanner *scanner) {
    return atoll(LightScanner_string(scanner));
}

void LightScanner_free(LightScanner *scanner) {
    free(scanner);
}

LightWriter* LightWriter_new(FILE *out) {
    LightWriter *writer = (LightWriter*)malloc(sizeof(LightWriter));
    writer->out = out;
    return writer;
}

void LightWriter_print(LightWriter *writer, const char *s) {
    fprintf(writer->out, "%s", s);
}

void LightWriter_ans(LightWriter *writer, long long l) {
    fprintf(writer->out, "%lld", l);
}

void LightWriter_ansln(LightWriter *writer, long long n) {
    LightWriter_ans(writer, n);
    fprintf(writer->out, "\n");
}

void LightWriter_close(LightWriter *writer) {
    fclose(writer->out);
    free(writer);
}

int lowerBound(long long *a, int n, long long t) {
    int min = 0, max = n;
    while (min < max) {
        int mid = (min + max) / 2;
        if (t <= a[mid]) {
            max = mid;
        } else {
            min = mid + 1;
        }
    }
    return min;
}

long long IntMath_min(long long a, long long b, long long c, long long d, long long e, long long f) {
    long long min_val = a;
    if (b < min_val) min_val = b;
    if (c < min_val) min_val = c;
    if (d < min_val) min_val = d;
    if (e < min_val) min_val = e;
    if (f < min_val) min_val = f;
    return min_val;
}

void DLazyFaith_solve(int testNumber, LightScanner *in, LightWriter *out) {
    int a = LightScanner_ints(in), b = LightScanner_ints(in), q = LightScanner_ints(in);
    long long s[MAX_N + 2], t[MAX_N + 2];
    s[0] = t[0] = -INF;
    for (int i = 0; i < a; i++) {
        s[i + 1] = LightScanner_longs(in);
    }
    for (int i = 0; i < b; i++) {
        t[i + 1] = LightScanner_longs(in);
    }
    s[a + 1] = t[b + 1] = INF;
    for (int i = 0; i < q; i++) {
        long long x = LightScanner_longs(in);
        long long sl = x - s[lowerBound(s, a + 2, x + 1) - 1];
        long long tl = x - t[lowerBound(t, b + 2, x + 1) - 1];
        long long sr = s[lowerBound(s, a + 2, x)] - x;
        long long tr = t[lowerBound(t, b + 2, x)] - x;
        LightWriter_ansln(out, IntMath_min(sl, tl, sr, tr, 2 * sl + tr, 2 * tl + sr));
    }
}

int main() {
    LightScanner *in = LightScanner_new(stdin);
    LightWriter *out = LightWriter_new(stdout);
    DLazyFaith_solve(1, in, out);
    LightWriter_close(out);
    LightScanner_free(in);
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/Go/s041542819_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	in := NewLightScanner(os.Stdin)
	out := NewLightWriter(os.Stdout)
	defer out.Close()

	solver := &DLazyFaith{}
	solver.Solve(1, in, out)
}

type DLazyFaith struct{}

func (d *DLazyFaith) Solve(testNumber int, in *LightScanner, out *LightWriter) {
	a := in.Int()
	b := in.Int()
	q := in.Int()
	s := make([]int64, a+2)
	t := make([]int64, b+2)
	s[0], t[0] = -10000000000, -10000000000
	for i := 0; i < a; i++ {
		s[i+1] = in.Long()
	}
	for i := 0; i < b; i++ {
		t[i+1] = in.Long()
	}
	s[a+1], t[b+1] = 20000000000, 20000000000
	for i := 0; i < q; i++ {
		x := in.Long()
		sl := x - s[max(0, lowerBound(s, x+1)-1)]
		tl := x - t[max(0, lowerBound(t, x+1)-1)]
		sr := s[lowerBound(s, x)] - x
		tr := t[lowerBound(t, x)] - x
		out.Ansl(min(
			max(sl, tl),
			max(sr, tr),
			2*sl+tr,
			2*tl+sr,
			sl+2*tr,
			tl+2*sr,
		))
	}
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(values ...int64) int64 {
	minValue := values[0]
	for _, v := range values {
		if v < minValue {
			minValue = v
		}
	}
	return minValue
}

func lowerBound(a []int64, t int64) int {
	min, max := 0, len(a)
	for min < max {
		mid := (min + max) / 2
		if t <= a[mid] {
			max = mid
		} else {
			min = mid + 1
		}
	}
	return min
}

type LightScanner struct {
	reader    *bufio.Reader
	tokenizer *strings.FieldsFunc
}

func NewLightScanner(r io.Reader) *LightScanner {
	return &LightScanner{
		reader: bufio.NewReader(r),
	}
}

func (l *LightScanner) String() string {
	if l.tokenizer == nil {
		l.tokenizer = l.nextTokenizer()
	}
	if len(*l.tokenizer) == 0 {
		*l.tokenizer = l.nextTokenizer()
	}
	token := (*l.tokenizer)[0]
	*l.tokenizer = (*l.tokenizer)[1:]
	return token
}

func (l *LightScanner) nextTokenizer() []string {
	var buffer strings.Builder
	for {
		line, isPrefix, err := l.reader.ReadLine()
		if err != nil {
			panic(err)
		}
		buffer.Write(line)
		if !isPrefix {
			break
		}
	}
	return strings.FieldsFunc(buffer.String(), func(r rune) bool {
		return r == ' ' || r == '\t' || r == '\n'
	})
}

func (l *LightScanner) Int() int {
	i, err := strconv.Atoi(l.String())
	if err != nil {
		panic(err)
	}
	return i
}

func (l *LightScanner) Long() int64 {
	i, err := strconv.ParseInt(l.String(), 10, 64)
	if err != nil {
		panic(err)
	}
	return i
}

type LightWriter struct {
	writer *bufio.Writer
}

func NewLightWriter(w io.Writer) *LightWriter {
	return &LightWriter{
		writer: bufio.NewWriter(w),
	}
}

func (l *LightWriter) Print(a interface{}) {
	_, err := fmt.Fprintf(l.writer, "%v", a)
	if err != nil {
		panic(err)
	}
}

func (l *LightWriter) Ansl(a ...interface{}) {
	for i, v := range a {
		if i > 0 {
			l.Print(" ")
		}
		l.Print(v)
	}
	l.Println()
}

func (l *LightWriter) Println() {
	l.Print("\n")
}

func (l *LightWriter) Close() {
	err := l.writer.Flush()
	if err != nil {
		panic(err)
	}
}

// End of Code
